rust/system/src/execution/orchestrator.rs

```diff
@@ -38,8 +41,13 @@ pub trait Orchestrator: Debug + Send + Sized + 'static {
     }
 
     /// Sends a task to the dispatcher and return whether the task is successfully sent
-    async fn send(&mut self, task: TaskMessage, ctx: &ComponentContext<Self>) -> bool {
-        let res = self.dispatcher().send(task, Some(Span::current())).await;
+    async fn send(
+        &mut self,
+        task: TaskMessage,
+        ctx: &ComponentContext<Self>,
+        tracing_context: Option<Span>,
```

Comment:
this signature makes it a bit annoying for the 95% of usecases that want to propagate the same trace tree
maybe people who want a new root should just call `self.dispatcher().send()` directly instead?

Reply:
I am strongly opposed to this change and agree with @codetheweb here. This smells bad. I would appreciate we took time with changes to critical framework code.

Reply:
Not able to understand why the orchestrator passing the trace tree is bad. It is the planner so it is in charge of deciding the DAG layout and so feels like it should be in charge of passing the trace tree (which is dependent on the DAG) as well

Reply:
For e.g. if I want to group fetch posting list operator with the corresponding brute force posting list operator as siblings under the parent spann of orchestrator then only the orchestrator can capture such a relationship. The operators themselves wouldn't know how do it without any extra context